app/services/creator_service.py

```diff
@@ -202,3 +202,16 @@ async def post_async_profile_analytics(request_body: object, params: Optional[Di
         await executor_event.async_profile_analytics_request_event_handler(data=async_profile_analytics)
 
     return async_profile_analytics
+
+
+async def post_async_contents_fetch(request_body: object, params: Optional[Dict]) -> Optional[Dict]:
```

Comment:
this api should raise an exception if `Product.PUBLIC_CONTENT_SEARCH` not in `settings.SUPPORTED_PRODUCTS`

same needs to be done for profile analytics and other product dependent APIs